Python/specialize.c

```diff
@@ -1212,6 +1262,36 @@ function_kind(PyCodeObject *code) {
     return SIMPLE_FUNCTION;
 }
 
+/* Returning 0 indicates a failure. */
+static uint32_t
+function_get_version(PyObject *o, int expected_argcount, int opcode)
+{
+    uint32_t version = 0;
+    if (!Py_IS_TYPE(o, &PyFunction_Type)) {
```

Comment:
This should go back to `case PROPERTY:` the `case GET_CALL_GETATTRIBUTE:` already asserts that `o` is a function.
Also the name `function_get_version` implies that `o` must be a function.